batchflow/models/torch/layers/conv_complex.py

```diff
@@ -338,3 +341,84 @@ def __init__(self, inputs=None, kernel_size=(3, 5, 7), stride=1, dilation=1, gro
 
     def forward(self, x):
         return Combine(op=self.combine)([layer(x) for layer in self])
+
+    
+
+class DeformableConv2d(nn.Module):
+    """ Deformable convolution: apply convolution with learnable grid sampling locations and feature amplitudes modulation.
+    Grid offsets and modulation scalars are learned with separate convolution layers.
+    Wang, Wenhai, et al. "`InternImage: Exploring Large-Scale Vision Foundation Models with Deformable Convolutions
+    <https://arxiv.org/abs/2211.05778>`_"
+
+    Parameters
+    ----------
+    offset_groups : int
+        `groups` for offset and modulation convolution layers. Default is 1.
+    """
+    LAYER = nn.Conv2d
```

Comment:
As this layer is 2D only (and I doubt that `PyTorch` would add 3D implementation any soon), I would not use this class attribute and just hardcode `Conv2d` calls